Dockerfile

```diff
@@ -1,32 +1,36 @@
 FROM quay.io/ukhomeofficedigital/centos-base
 
-ENV RUBY_VERSION 2.3.0
+ENV RUBY_VERSION 2.3.3
 
-RUN yum clean all && \
-    yum update -y && \
-# Install all packages required by RVM
-    yum install -y which patch libyaml-devel \
-    glibc-headers autoconf gcc-c++ glibc-devel \
-    patch readline-devel zlib-devel libffi-devel \
-    openssl-devel make bzip2 automake libtool \
-    bison sqlite-devel && \
-    yum clean all && \
-    rpm --rebuilddb
+# Onbuild commands ensure packages are updated whenever an image is built on top of this one
+ONBUILD RUN yum clean all && \
+            yum update -y && \
+            yum install -y git && \
+            yum clean all && \
+            rpm --rebuilddb
+ONBUILD ADD Gemfile /app
+ONBUILD RUN bundle install
+ONBUILD ADD . /app
 
-#Install RVM with latest stable ruby and bundle
-RUN curl -sSL https://rvm.io/mpapis.asc | gpg2 --import -  && \
-    /usr/bin/curl -L https://get.rvm.io | /bin/bash -s stable --ruby \
-    --with-default-gems="bundle"
+RUN yum install -y wget make bzip2 gcc-c++
+
+#   Install Ruby
+RUN \
+#   Get GPG key
+    wget https://raw.github.com/postmodern/postmodern.github.io/master/postmodern.asc && \
+    gpg --import postmodern.asc && \
+    wget https://raw.github.com/postmodern/ruby-install/master/pkg/ruby-install-0.6.0.tar.gz.asc && \
+#   Download ruby-install and verify against the GPG key
+    wget -O ruby-install-0.6.0.tar.gz https://github.com/postmodern/ruby-install/archive/v0.6.0.tar.gz && \
+    gpg --verify ruby-install-0.6.0.tar.gz.asc ruby-install-0.6.0.tar.gz && \
+    tar -xzvf ruby-install-0.6.0.tar.gz && cd ruby-install-0.6.0 && make install && \
+#    Install Ruby
+    ruby-install --system ruby 2.2.6 && gem install bundler
 
 #Create app user and group
 RUN groupadd -r app && \
     useradd -r -g app app -d /app && \
     mkdir -p /app && \
     chown -R app:app /app
 
-ENV GEM_PATH=/usr/local/rvm/gems/ruby-${RUBY_VERSION}:/usr/local/rvm/gems/ruby-${RUBY_VERSION}@global \
-    RUBY_VERSION=ruby-${RUBY_VERSION} \
-    GEM_HOME=/usr/local/rvm/gems/ruby-${RUBY_VERSION} \
-    PATH=${PATH}:/usr/local/rvm/rubies/ruby-${RUBY_VERSION}/bin:/usr/local/rvm/gems/ruby-${RUBY_VERSION}/bin:/usr/local/rvm/gems/ruby-${RUBY_VERSION}@global/bin
-
 WORKDIR /app
```

Comment:
@Joseph-Irving you can't set the user in this Dockerfile as for the yum update etc to work with the onbuild command it'll still need to be root